### DISCUSSION TITLE
FairyGUI RichText AutoSize is not working as expected

### DISCUSSION THREAD

**Discussion by iAbdulQadirKhan** [axmolengine/axmol#1598] [General]
This works fine if RichText's **AutoSize** is None. If **AutoSize** is Height + Width then it doesn't work as expected.

**Expected:**
<img width="597" alt="Screenshot 2024-01-11 at 2 11 20 PM" src="https://github.com/axmolengine/axmol/assets/79637143/88b0fea8-ffd2-4874-bc6f-dd08295a283a">

**Actual:**
<img width="655" alt="Screenshot 2024-01-11 at 2 11 34 PM" src="https://github.com/axmolengine/axmol/assets/79637143/86876403-2335-4f91-b48c-39545df7d659">

**FairyGUI Design:**
<img width="1274" alt="Screenshot 2024-01-11 at 1 57 33 PM" src="https://github.com/axmolengine/axmol/assets/79637143/87f4deaa-57d8-4082-b881-2f0d236391f2">

**Comment by iAbdulQadirKhan** [comment]
In **cocos2d-v3** it was working fine with **AutoSize** Height + Width but now it converts multilines text into single line.

**Reply by rh101** [reply]
You would have more luck receiving help if you post up a test project with the least amount of code to reproduce the issue.

**Reply by iAbdulQadirKhan** [reply]
I have created a component in fairyGUI. In this component I have a rich text. And I'm just showing this component in game.

```
<?xml version="1.0" encoding="utf-8"?>
<component size="2048,1152" designImageAlpha="35">
  <displayList>
    <richtext id="n55_xe4g" name="txt_01" xy="1002,546" pivot="0.5,0.5" anchor="true" size="1332,294" scale="0.8,0.8" touchable="false" font="ui://a0gz3e6xprke0" fontSize="12" color="#ffffff" align="center" vAlign="middle" text="Tap on a chip to select&#xA;a bet amount"/>
  </displayList>
</component>
```

**Comment by iAbdulQadirKhan** [comment]
I have changed the parser option from HTML to Normal. This change fixes the multilines problem but creates an alignment problem.

**change:**
<img width="744" alt="Screenshot 2024-01-17 at 7 23 18 PM" src="https://github.com/axmolengine/axmol/assets/79637143/d8a2c5d4-f86d-49e1-b7e8-daef64e4151f">

**result:**
<img width="313" alt="Screenshot 2024-01-17 at 7 19 01 PM" src="https://github.com/axmolengine/axmol/assets/79637143/73b130c2-cf82-4cf4-89f8-0f9a96149c24">

**Reply by rh101** [reply]
You stated that the functionality was working in cocos2d-x v3, so if that is the case, then why not run both a cocos2d-x v3 build and an Axmol build side-by-side, and step through the code that parses and formats the text?  You could then figure out exactly where the differences in the output are.  Strategic use of breakpoints would save you stepping through irrelevant sections code to save you time.

**Reply by iAbdulQadirKhan** [reply]
I have compared the richtext related code with the cocos2d-v3 but not find the solution of this problem.

**Reply by rh101** [reply]
Comparing the code is not enough. As I mentioned earlier, you need to run the code for both cocos2d-x v3 and Axmol, and step through through them with the debugger to find where the the output differs.

**Comment by iAbdulQadirKhan** [comment]
Sorry I was wrong. The alignment issue is also present in cocos2d-v3. The reason for the above change is that the html parser removes the escaped characters (\n) from the text string.
